cdap-app-fabric/src/main/java/co/cask/cdap/test/internal/AppFabricClient.java

```diff
@@ -322,4 +318,16 @@ public static File createDeploymentJar(LocationFactory locationFactory, Class<?>
     return new File(deployJar.toURI());
   }
 
-}
+  public static File createDeploymentJar(LocationFactory locationFactory, Class<?> clz, File...bundleEmbeddedJars)
+    throws IOException {
+    // Creates Manifest
+    Manifest manifest = new Manifest();
+
+    Attributes attributes = new Attributes();
+    attributes.put(ManifestFields.MAIN_CLASS, clz.getName());
+    attributes.put(ManifestFields.MANIFEST_VERSION, "1.0");
+    manifest.getMainAttributes().putAll(attributes);
+
+    return createDeploymentJar(locationFactory, clz, manifest);
```

Comment:
Yes. I did not want to make this PR even longer. It is already a bit large. More tests will be added in subsequent PRs. Which will include bundled jar.